opt/se-logger/liveupdate.py

```diff
@@ -113,23 +114,24 @@ def __init__(self, key, msg0503):
                  parameters 0239, 023a, 023b, and 023c.
         msg0503: a 34-byte string with the contents of a 0503 message.
         """
-        enkey1 = map(ord, AES.new(key).encrypt(msg0503[0:16]))
-        self.cipher = AES.new("".join(map(chr,
-            (enkey1[i] ^ ord(msg0503[i+16]) for i in range(16)))))
+        enkey1 = list(map(lambda x: ord(chr(x)), AES.new(key, AES.MODE_ECB).encrypt(msg0503[0:16])))
+        if len(enkey1) == 0:
+            return
+        self.cipher = AES.new(bytes(enkey1[i] ^ ord(chr(msg0503[i+16])) for i in range(16)), AES.MODE_ECB)
 
     def decrypt(self, msg003d):
         """
         msg003d: the contents of the 003d message to decrypt.
 
         Returns a tuple(int(sequenceNumber), string(data)).
         """
-        rand1 = map(ord, msg003d[0:16])
-        rand = map(ord, self.cipher.encrypt(msg003d[0:16]))
-        msg003d = map(ord, msg003d)
+        rand1 = list(map(lambda x: ord(chr(x)), msg003d[0:16]))
+        rand = list(map(lambda x: ord(chr(x)), self.cipher.encrypt(msg003d[0:16])))
+        msg003d = list(map(lambda x: ord(chr(x)), msg003d))
```

Comment:
```suggestion
        rand1 = msg003d[0:16]
        rand = self.cipher.encrypt(bytes(msg003d[0:16]))
```

Reply:
Ah, so it becomes mutable instead of bytes()?

Reply:
Yes!